[CROM-6661][BT-114] Reference Disk Opt-in: UI work

mcovarr opened the pull request:
Confirmed with Chrome network traffic monitoring that the `useReferenceDisks` parameter was being sent in submission requests as expected.

mcovarr committed:
wip

mcovarr committed:
wip

panentheos reviewed (approved):
Looks good. Tested locally by verifying the flag in the network call.

kshakir commented on src/pages/workspaces/workspace/workflows/WorkflowView.js:
```diff
@@ -763,6 +764,13 @@ const WorkflowView = _.flow(
+                checked: useReferenceDisks,
+                onChange: v => this.setState({ useReferenceDisks: v }),
+                style: { marginLeft: '1rem' }
+              }, [' Use reference disks'])
```
Is that space before `Use` supposed to be there?

panentheos replied:
Yes, the `LabeledCheckbox` component puts the checkbox inline with the text, so we typically put a space to create a small gap between them.

grsterin reviewed (approved):
Looks good, once Khalid's comment is answered

mcovarr committed:
add info box with dummy link, real link pending

mcovarr committed:
correct link

mcovarr commented:
Added info box with link to Terra Support article. I also had to do a nontrivial rebase, confirmed the 'useReferenceDisks' parameter is still working per network traffic.

breilly2 reviewed (approved):
👍  Looks good. I tested locally by verifying that the flag shows up in the call to Rawls, both on initial submission and when rerunning failures.

mcovarr committed:
not sure if this got killed in a rebase or it was left out in the original... 😬

mcovarr commented:
Good news: I fixed the integration test issues with a rebase.

Less good news: Said rebase was not completely trivial. Also in the process of testing I noticed that 'Use Reference Disks' was missing from `SubmissionDetails.js` so I added that in. All of which means I must ask for another round of review... 😞

breilly2 reviewed (approved):
👍   Still looks good. I verified that "Use Reference Disks" shows up on the job details screen.

BTW, the test failure was more likely fixed by a service account clean-up I did yesterday. See this slack thread: https://broadinstitute.slack.com/archives/C01EHNUM73R/p1612885718027100
